service_discovery/proto/chariott/v1/chariott_registry.proto

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+syntax = "proto3";
+package chariott_registry;
+
+// Service Registry definition
+//
+// The protobuf definitions for Chariott's service registry
+
+// The entry point for the Registry gRPC Service.
+service Registry {
+  // Register, or add a service to the registry
+  rpc Register(RegisterRequest) returns (RegisterResponse) {}
+
+  // Unregister, or remove a service from the registry
+  rpc Unregister(UnregisterRequest) returns (UnregisterResponse) {}
+
+  // Discover, or retrieve the metadata for a single service given its fully qualified name
+  rpc DiscoverService(DiscoverServiceRequest) returns (DiscoverServiceResponse) {}
```

Comment:
what about naming it just Discover; as this is a registry of services it is implied that it is about discovering services; for example we do not have RegisterService

Reply:
Will change. I had it "DiscoverService" to differentiate from "DiscoverByNamespace" but I agree that Discover matches Register better